src/CompiledExpression.php

```diff
@@ -294,7 +294,7 @@ public function isScalar()
      */
     public function isArray()
     {
-        return $this->type === self::ARR;
+        return ($this->type === self::ARR && is_array($this->value));
```

Comment:
Hey!

You should find problem in another place, code should equals

$variable->isArray() && $variable->isCorrectValue()

I will explain:
Some times PHPSA known that type is array but dont known value
And value will be null

Thank